src/hooks/sync-sites/use-site-sync-management.ts

```diff
@@ -37,6 +39,33 @@ export const useSiteSyncManagement = ( {
 		}
 	}, [ isAuthenticated, loadConnectedSites ] );
 
+	// whenever array of syncSites changes, we need to update connectedSites to keep them updated with wordpress.com
+	useEffect( () => {
+		if ( isFetching || ! isAuthenticated ) {
+			return;
+		}
+
+		setConnectedSites( ( prevConnectedSites ) => {
+			const updatedConnectedSites = prevConnectedSites.map( ( connectedSite ) => {
+				const site = syncSites.find( ( site ) => site.id === connectedSite.id );
+
+				if ( ! site ) {
+					return connectedSite;
+				}
+
+				return {
+					...connectedSite,
+					syncSupport: site.syncSupport,
+					url: site.url,
+				};
+			} );
+
+			getIpcApi().updateConnectedWpcomSites( updatedConnectedSites );
```

Comment:
I'm not totally convinced about calling an effect to update the appData inside a setState. I know that we are doing the same in other parts of the code, and it's easy to read, but probably not the best react approach.